### PR TITLE
Improving serialization interface

### PR DESCRIPTION
**Pull request by corujautx**
Changing `serialize(data: Data?, error: Swift.Error?)` to `serialize(response: Result<Data>)` so it becomes easier (and stylish) to write custom parsers (e.g. using `UnboxableWithContext` interface).

### CODE REVIEW

File: Source/Networking/HTTPClient.swift [hunk 1]
```diff
@@ -32,7 +32,13 @@ public extension HTTPClient {
     }
 }
 
+enum HTTPResourceErrors: Error {
```
Comment (fellipecaetano) [inline Source/Networking/HTTPClient.swift]: I think it's best if this is called `HTTPResourceError`. What do you think?

**Review by fellipecaetano — CHANGES_REQUESTED** [review]
Except for a type name that I think should be changed, everything is great 💯

File: Source/Networking/HTTPClient.swift (hunk 1 above)
Comment (corujautx) [reply Source/Networking/HTTPClient.swift]: I'll change it.

**Comment by corujautx** [comment]
@fellipecaetano please check the changes

**Review by fellipecaetano — APPROVED** [review]

**Pull request merged**